pixl_core/src/core/uploader/_dicomweb.py

```diff
@@ -36,82 +38,119 @@ def __init__(self, project_slug: str, keyvault_alias: Optional[str]) -> None:
-        msg = "Currently not implemented. Use `send_via_stow()` instead."
-        raise NotImplementedError(msg)
-
-    def send_via_stow(self, resource_id: str) -> requests.Response:
+        self.az_prefix = az_prefix if az_prefix else self.project_slug
+
+        self.orthanc_user = config("ORTHANC_ANON_USERNAME")
+        self.orthanc_password = config("ORTHANC_ANON_PASSWORD")
+        self.orthanc_url = config("ORTHANC_ANON_URL")
+        self.endpoint_user = self.keyvault.fetch_secret(f"{self.az_prefix}--dicomweb--username")
+        self.endpoint_password = self.keyvault.fetch_secret(f"{self.az_prefix}--dicomweb--password")
+        self.endpoint_url = self.keyvault.fetch_secret(f"{self.az_prefix}--dicomweb--url")
+        # The DICOMweb API endpoint on the Orthanc server, used by Orthanc to interact with the
+        # DICOMweb server. Note that this is different from the endpoint_url, which is the URL of
+        # the DICOMweb server itself.
+        self.orthanc_dicomweb_url = self.orthanc_url + "/dicom-web/servers/" + self.az_prefix
+        self.http_timeout = int(config("HTTP_TIMEOUT", default=30))
+
+    def upload_dicom_image(self, study_id: str) -> None:
+        pseudo_anon_image_id, _ = get_tags_by_study(study_id)
+        logger.info("Starting DICOMweb upload of '{}'", pseudo_anon_image_id)
+        self.send_via_stow(study_id, pseudo_anon_image_id)
+        logger.info("Finished DICOMweb upload of '{}'", pseudo_anon_image_id)
+
+    def send_via_stow(self, resource_id: str, pseudo_anon_image_id: str) -> requests.Response:
         """Upload a Dicom resource to the DicomWeb server from within Orthanc."""
-        if not self._check_dicomweb_server():
+        if not self._check_dicomweb_server_exists():
             logger.info("Creating new DICOMWeb credentials")
             self._setup_dicomweb_credentials()
 
-        headers = {"content-type": "application/dicom", "accept": "application/dicom+json"}
+        self._validate_dicomweb_server()
+        super().check_already_exported(pseudo_anon_image_id)
+
+        headers = {"content-type": "application/json", "accept": "application/dicom+json"}
         payload = {"Resources": [resource_id], "Synchronous": False}
 
         try:
             response = requests.post(
-                self.url + "/stow",
-                auth=(self.user, self.password),
+                self.orthanc_dicomweb_url + "/stow",
+                auth=(self.orthanc_user, self.orthanc_password),
                 headers=headers,
                 data=json.dumps(payload),
-                timeout=30,
+                timeout=self.http_timeout,
             )
             response.raise_for_status()
         except requests.exceptions.RequestException:
             logger.error("Failed to send via stow")
             raise
         else:
+            super().update_exported_timestamp(pseudo_anon_image_id)
             logger.info("Dicom resource {} sent via stow", resource_id)
```

Comment:
else block not needed
```suggestion
        super().update_exported_timestamp(pseudo_anon_image_id)
        logger.info("Dicom resource {} sent via stow", resource_id)
```